### ISSUE TITLE
Add useThemedStyle hook

### ISSUE THREAD

**Issue by colingourlay** [colingourlay/styled-hooks#5]
This will use a `$prop` syntax to do the same as `useTheme()` followed by ```useStyle`` ```.

e.g.:

```js
const { textColor } = useTheme();
const className = useStyle`
  color: ${textColor}
`;
```

...can be acheived with:

```js
const className = useThemedStyle`
  color: $textColor
`;
```

This should allow access to nested theme props & array access for things style systems, e.g:

```js
const { sizes } = useTheme();
const className = useStyle`
  margin: ${sizes[2]}
`;
```

...should also be possible with:

```js
const className = useThemedStyle`
  color: $sizes.2
`;
```

**Comment by colingourlay** [comment]
This should probably stick to the familiar SASS/Ruby interpolation pattern for props...

```js
const className = useThemedStyle`
  color: #{textColor};
  margin: #{sizes.2};
`;
```

This has the advantage of (for most cases) being a single character change between JS's template string interpolation and theme interpolation. Sure it's a couple of characters longer than the initial proposed pattern, but I think braces are clearer.

**Comment by colingourlay** [comment]
Closed by #12
